handler/turtle.py

```diff
@@ -0,0 +1,274 @@
+        self.metadata_file = os.path.join(self.local_path,
+                                          config.get('METADATA_FILE'))
+        self.collection    = pcdm.Collection.from_repository(repo,
+                                                     config.get('COLLECTION'))
+
+        # Create data structures to accumulate process results
+        self.items       = {}
+        self.incomplete  = []
+        self.extra_files = []
+
+        # Check for required metadata file and path
+        if not os.path.isdir(self.metadata_path):
+            os.mkdir(self.metadata_path)
+        if not os.path.isfile(self.metadata_file):
+            raise ConfigException('Specified metadata file could not be found')
+
+        # Generate index of all files in the data path
+        self.logger.info('Walking the "data path" tree to create a file index')
+        self.all_files = {}
+        for root, dirs, files in os.walk(self.data_path):
+            for f in files:
+                self.all_files[f] = os.path.join(root,f)
+        self.logger.info("Found {0} files".format(len(self.all_files)))
+
+        # Generate item-level metadata graphs and store as files
+        with open(self.metadata_file, 'r') as f:
+            self.logger.info('Parsing the master metadata graph')
+            g = Graph().parse(f, format="turtle")
+            # For each of the unique subjects in the graph
+            for subj_uri in set([i for i in g.subjects()]):
+                # Get the item identifier
+                itembase = os.path.basename(subj_uri)
+                # Create the path to the output file
+                outfile = os.path.join(self.metadata_path, itembase) + '.ttl'
+                # Create a graph of all triples with that subject
+                itemgraph = Graph()
+                itemgraph += g.triples((subj_uri, None, None))
+                # Serialize the graph to the path location
+                self.logger.info('Serializing graph {0}'.format(outfile))
+                itemgraph.serialize(destination=outfile, format="turtle")
+
+        # If available, read batch index from file
+        if os.path.isfile(self.index_file):
+            self.logger.info("Reading batch index from {0}".format(self.index_file))
+            with open(self.index_file, 'r') as infile:
+                self.items = yaml.load(infile)
+
+        # Otherwise, construct the index by reading graph files
+        else:
+            for f in os.listdir(self.metadata_path):
+                fullpath = os.path.join(self.metadata_path, f)
+                # skip files starting with dot
+                if f.startswith('.'):
+                    self.extra_files.append(fullpath)
+                    continue
+                else:
+                    try:
+                        id = os.path.basename(f).rstrip('.ttl')
+                        self.items[id] = self.create_item(fullpath)
+                    except:
```

Comment:
Having this bare `except` here is simply swallowing any errors without logging any sort of explanation. It would be better to explicitly catch the anticipated exceptions (e.g., `KeyError` when `create_item` cannot find a file in the index) and issue a warning via the logger.